### PR TITLE
system/service.py: fix false-positive service enablement on FreeBSD.

### PR DESCRIPTION
**Pull request by problame**
This is follow-up to https://github.com/ansible/ansible-modules-core/pull/4042:

I did all my tests for the as `root`, where sysrc(8) and the original patch worked flawlessly.

However, sysrc(8) **does not return with non-zero exit status** on a permission error.
### Demo

```
# User vagrant does not have write access to rc.conf, yet this reports successful change 
ansible -i inventory -u vagrant -m service -a 'name=sshd enabled=no' testvm
```
### Mitigation

Mitigations I considered before writing this patch:
1. Revert the commit, re-introducing https://github.com/ansible/ansible-modules-core/issues/2664 and effectively not parsing `rc.conf` correctly (see the linked issue)
2. Check in the module code whether current user is `root` (is there a helper function?). Hard-codes the assumption that only root can edit `rc.conf`, which is how it should be in most cases.
3. Check if the current user can write to `rc.conf` (is there a helper function?). Problem: hard-codes the assumption that the data will be stored in `rc.conf`.
4. Check if `stderr` is not empty. This works for now, but may break in the future.
5. Use service(8) `service <name> enabled` to check if the service is in fact enabled/disabled after calling sysrc(8). service(8) is also used to determine the `rcvar` to set, hence validating the change with service(8) makes sense. 

As you can see in this PR, I went for option 5.

I apologize for the incomplete testing and hope the changes will be merged soon.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @problame. To the current maintainers, @grossws please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by grossws** [comment]
shipit

**Comment by grossws** [comment]
@problame , patch seems ok to me, thank you for quick fix

**Comment by gregdek** [comment]
Thanks again to @problame for this PR, and thanks @grossws for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @problame for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

File: system/service.py [hunk 1]
```diff
@@ -1005,9 +1005,15 @@ def service_enable(self):
...
 
-                rc, stdout, stderr = self.execute_command("%s %s=\"%s\"" % (self.sysrc_cmd, self.rcconf_key, self.rcconf_value ) )
+                rc, change_stdout, change_stderr = self.execute_command("%s %s=\"%s\"" % (self.sysrc_cmd, self.rcconf_key, self.rcconf_value ) )
                 if rc != 0:
-                    self.module.fail_json(msg="unable to set rcvar using sysrc", stdout=stdout, stderr=stderr)
+                    self.module.fail_json(msg="unable to set rcvar using sysrc", stdout=change_stdout, stderr=change_stderr)
+
+                # sysrc does not exit with code 1 on permission error => validate successful change using service(8)
+                rc, check_stdout, check_stderr = self.execute_command("%s %s %s" % (self.svc_cmd, self.name, "enabled"))
+                is_enabled = True if rc == 0 else False
```
Comment (mattclay) [inline system/service.py]: This condition needs to be rewritten, as it doesn't work on python 2.4.

As a result, the tests on Travis fail:

```
$ python2.4 -m compileall -fq -x 'cloud/' .
Compiling ./system/service.py ...
  File "./system/service.py", line 1014
    is_enabled = True if rc == 0 else False
                       ^
SyntaxError: invalid syntax
```

Comment (problame) [inline system/service.py]: Fixed.

**Comment by grossws** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @problame for this PR, and thanks @grossws for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**